src/Illuminate/Queue/Worker.php

```diff
@@ -160,7 +160,11 @@ protected function registerTimeoutHandler($job, WorkerOptions $options)
         pcntl_signal(SIGALRM, function () use ($job, $options) {
             if ($job) {
                 $this->markJobAsFailedIfWillExceedMaxAttempts(
-                    $job->getConnectionName(), $job, (int) $options->maxTries, $this->maxAttemptsExceededException($job)
+                    $job->getConnectionName(), $job, (int) $options->maxTries, $e = $this->maxAttemptsExceededException($job)
```

Comment:
Should this not being another exception? Something more "clear" about the purpose, a timeout? The message of this exception is not clear about the reason for the exception but here we know it's a timeout.

Reply:
We don't know it's a timeout. The job may have thrown an exception several times and then timedout.

Reply:
This is the timeoutHandler, so a timeoutException should be better than `MaxAttemptsExceededException`? There are only 2 places where that exception is thrown: markJobAsFailedIfAlreadyExceedsMaxAttempts (that's ok), and registerTimeoutHandler which has nothing to do with attempts? Only if the exception that is thrown hits the amount of retries. Instead of re-using the created on from the lines above, it should be better to throw another specific TimeoutException, the one above can than be simplified without the mentioning of the timeout.